# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Tenuring.cpp
# Commit: 72bf9c871ded
# Full Hash: 72bf9c871ded006f56f09b24d0723365516445bb
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 5
# Description:
#   Bug 1692065 - wasm: Add tagging for 31-bit integers and i31ref. r=yury
#   
#   This commit adds a tag 0x1 for 31-bit integers in AnyRef. Any JS value
#   that is losslessly representable (without coercion) in this range is
#   stored without boxing. This is chosen to line up exactly with how the
# ==============================================================================

diff -r 28feda14fdec -r 72bf9c871ded js/src/gc/Tenuring.cpp
--- a/js/src/gc/Tenuring.cpp	Fri Aug 04 14:06:50 2023 +0000
+++ b/js/src/gc/Tenuring.cpp	Fri Aug 04 14:06:50 2023 +0000
@@ -189,6 +189,7 @@
       post = wasm::AnyRef::fromJSString(str);
       break;
     }
+    case wasm::AnyRefKind::I31:
     case wasm::AnyRefKind::Null: {
       // This function must only be called for GC things.
       MOZ_CRASH();